core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -247,47 +251,65 @@ public Object visitRollbackCommand(TxInvocationContext ctx, RollbackCommand comm
-      boolean syncCommitPhase = cacheConfiguration.transaction().syncCommitPhase();
-      RpcOptions rpcOptions;
-      if (syncCommitPhase) {
-         rpcOptions = rpcManager.getRpcOptionsBuilder(ResponseMode.SYNCHRONOUS_IGNORE_LEAVERS, DeliverOrder.NONE).build();
-      } else {
-         rpcOptions = rpcManager.getDefaultRpcOptions(false, DeliverOrder.NONE);
-      }
-      Map<Address, Response> responseMap = rpcManager.invokeRemotely(recipients, command, rpcOptions);
-      checkTxCommandResponses(responseMap, command);
-   }
-
-   protected void checkTxCommandResponses(Map<Address, Response> responseMap, TransactionBoundaryCommand command) {
+   protected void checkTxCommandResponses(Map<Address, Response> responseMap, TransactionBoundaryCommand command,
+                                          LocalTxInvocationContext context, Collection<Address> recipients) {
+      boolean throwOutdatedTopologyException = false;
       for (Map.Entry<Address, Response> e : responseMap.entrySet()) {
          Address recipient = e.getKey();
          Response response = e.getValue();
-         if (response instanceof CacheNotFoundResponse) {
+         if (response == CacheNotFoundResponse.INSTANCE) {
             // No need to retry if the missing node wasn't a member when the command started.
             if (command.getTopologyId() == stateTransferManager.getCacheTopology().getTopologyId()
                   && !rpcManager.getMembers().contains(recipient)) {
-               log.tracef("Ignoring response from node not targeted %s", recipient);
+               if (trace) log.tracef("Ignoring response from node not targeted %s", recipient);
             } else {
-               log.tracef("Cache not running on node %s, or the node is missing", recipient);
-               throw new OutdatedTopologyException("Cache not running on node " + recipient);
+               if (checkResponseInPartitionHandling(command, context, recipients)) {
+                  if (trace) log.tracef("Cache not running on node %s, or the node is missing. It will be handled by the PartitionHandlingManager", recipient);
+                  return;
+               } else {
+                  if (trace) log.tracef("Cache not running on node %s, or the node is missing", recipient);
+                  throwOutdatedTopologyException = true;
+               }
             }
-         } else if (response instanceof UnsureResponse) {
-            log.tracef("Node %s has a newer topology id", recipient);
-            throw new OutdatedTopologyException("Cache not running on node " + recipient);
+         } else if (response == UnsureResponse.INSTANCE) {
+            if (trace) log.tracef("Node %s has a newer topology id", recipient);
+            throwOutdatedTopologyException = true;
+         }
+      }
+      if (throwOutdatedTopologyException) {
+         throw new OutdatedTopologyException("Cache not running on some recipients.");
```

Comment:
I think the exception message was actually wrong on the `UnsureResponse` branch, it should have been just like the log message.